File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/user/dto/UserInfo.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.user.dto;

import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.UserRole;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.Users;

public record UserInfo(String userEmail, String userNickname, UserRole userRole) {
    public static UserInfo fromEntity(Users user) {
        return new UserInfo(
                user.getUserEmail(),
                user.getUserNickname(),
                user.getUserRole()
        );
    }
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/user/dto/UserInfo.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.user.dto

import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.UserRole

data class UserInfo(
    val userEmail: String,
    val userNickname: String,
    val userRole: UserRole
)
